Refactor Function Controller (#8004)
kyma-project/kyma: components/function-controller/README.md

| ------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
| **APP_METRICS_ADDRESS**                           | Address on which controller metrics are exposed                                                                                                                                                                      | `:8080`                                                                                                                                                    |

> | **APP_LEADER_ELECTION_ENABLED**                   | Field that enables one instance of the Function Controller to manage the traffic among instances                                                                                                                                                                                                  | `false`                                                                                                                                                    |

Reviewer: ```suggestion
| **APP_LEADER_ELECTION_ENABLED**                   | Field that enables one instance of the Function Controller to manage the traffic among all instances                                                                                                                                                                                                  | `false`                                                                                                                                                    |
```

> | **APP_LEADER_ELECTION_ID**                        | Name of the ConfigMap that specifies the main instance of the Function Controller that manages the traffic                                                                                                                                                                  | `serverless-controller-leader-election-helper`                                                                                                             |

Reviewer: ```suggestion
| **APP_LEADER_ELECTION_ID**                        | Name of the ConfigMap that specifies the main instance of the Function Controller that manages the traffic among all instances                                                                                                                                                                  | `serverless-controller-leader-election-helper`                                                                                                             |
```

| **APP_CONFIG_WATCHER_ENABLE_CONTROLLERS**         | Field that enables Config Watcher controllers that propagate ConfigMaps and Secrets to all Namespaces                                                                                                                                                                                        | `true`                                                                                                                                                     |

> | **APP_CONFIG_WATCHER_BASE_NAMESPACE**             | Name of the Namespace with the serverless configuration (such as runtime, Secret and service account for the Docker Registry) propagated to other Namespaces                                                                                                                                                                    | `kyma-system`                                                                                                                                              |

Reviewer 2: `...service account` probably also with a capital letters.

Author: @kazydek ?

| **APP_FUNCTION_REQUEUE_DURATION**                 | Period of time after which the controller refreshes the status of a Function CR.                                                                                                                                          | `1m`                                                                                                                                                       |
| **APP_FUNCTION_BUILD_REQUESTS_CPU**               | Minimum amount of CPU assigned to the Job to build the function image. See [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document for available values.    | `350m`                                                                                                                                                     |

> | **APP_FUNCTION_BUILD_REQUESTS_MEMORY**            | Minimum amount of Memory assigned to the Job to build the function image. See [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document for available values. | `750mi`                                                                                                                                                    |

Reviewer: Could you mark somehow in the description the difference between `APP_FUNCTION_BUILD_REQUESTS_MEMORY` and `**APP_FUNCTION_BUILD_LIMITS_MEMORY**` (same goes for CPU) as they are exactly the same for now.

| **APP_FUNCTION_BUILD_LIMITS_CPU**                 | Maximum amount of CPU assigned to the Job to build the function image. See [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document for available values.    | `1`                                                                                                                                                        |
| **APP_FUNCTION_BUILD_LIMITS_MEMORY**              | Maximum amount of Memory assigned to the Job to build the function image. See [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document for available values. | `1Gi`                                                                                                                                                      |

> | **APP_FUNCTION_BUILD_RUNTIME_CONFIG_MAP_NAME**    | Name of the ConfigMap that contains runtime definition.                                                                                                                                                                   | `dockerfile-nodejs-12`                                                                                                                                     |

Reviewer: ```suggestion
| **APP_FUNCTION_BUILD_RUNTIME_CONFIG_MAP_NAME**    | Name of the ConfigMap that contains the runtime definition                                                                                                                                                                   | `dockerfile-nodejs-12`                                                                                                                                     |
```

> | **APP_FUNCTION_BUILD_EXECUTOR_IMAGE**             | Full mame of the Kaniko executor image.                                                                                                                                                                                   | `gcr.io/kaniko-project/executor:v0.17.1`                                                                                                                   |

Reviewer: ```suggestion
| **APP_FUNCTION_BUILD_EXECUTOR_IMAGE**             | Full name of the Kaniko executor image used for building lambda images and pushing them to the Docker Registry                                                                                                                                                                                | `gcr.io/kaniko-project/executor:v0.17.1`                                                                                                                   |
```

> | **APP_FUNCTION_BUILD_CREDS_INIT_IMAGE**           | Full name of the Credentials Initializer image.                                                                                                                                                                           | `gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/creds-init:v0.10.1`                                                                               |

Reviewer: ```suggestion
| **APP_FUNCTION_BUILD_CREDS_INIT_IMAGE**           | Full name of the Tekton image used for building lambdas. It is run in the Init Container to adjust the Docker credentials to the Kaniko-compatible format.                                                                                                                                                                           | `gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/creds-init:v0.10.1`                                                                               |
```

> | **APP_FUNCTION_DOCKER_ADDRESS**                   | Fully qualified domain name of the Docker Registry Service with port                                                                                                                                                      | `function-controller-docker-registry.kyma-system.svc.cluster.local:5000`                                                                                   |

Reviewer: ```suggestion
| **APP_FUNCTION_DOCKER_ADDRESS**                   | Internal address of the Docker Registry                                                                                                                                | `function-controller-docker-registry.kyma-system.svc.cluster.local:5000`                                                                                   |
```

> | **APP_FUNCTION_DOCKER_EXTERNAL_ADDRESS**          | Domain address through which connection to the Docker Registry is facilitated                                                                                                                                             | `registry.kyma.local`                                                                                                                                      |

Reviewer: ```suggestion
| **APP_FUNCTION_DOCKER_EXTERNAL_ADDRESS**          | External address of the Docker Registry                                                                | `registry.kyma.local`                                                                                                                                      |
```
